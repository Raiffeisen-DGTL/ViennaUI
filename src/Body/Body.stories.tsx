import React from 'react';
import { Story, Meta } from 'storybook';
import { Body, BodyProps } from './Body';
import { Heading } from '../Typography/Heading/Heading';
import { H1, H2, H3, H4, H5, P } from '../Typography';

export default {
    title: 'Development/Body',
    component: Body,
} as Meta;

export const Overview: Story<BodyProps> = (args) => {
    return <Body {...args}>This is content</Body>;
};
export const WithTypography: Story<BodyProps> = (args) => {
    return (
        <Body {...args}>
            <Heading color='currentColor'>Условия обработки персональных данных</Heading>
            <H1 color='currentColor'>Условия обработки персональных данных</H1>
            <H2 color='currentColor'>Условия обработки персональных данных</H2>
            <H3 color='currentColor'>Условия обработки персональных данных</H3>
            <H4 color='currentColor'>Условия обработки персональных данных</H4>
            <H5 color='currentColor'>Условия обработки персональных данных</H5>
            <P size='xxl' margin='xxl'>
                Клиент, сведения о котором содержатся в представленных в Банк документах, дает свое согласие на
                обработку АО «Райффайзенбанк» и подтверждает, что, давая такое согласие, действует своей волей и своем
                интересе.
            </P>
        </Body>
    );
};
WithTypography.storyName = 'С Typography';
