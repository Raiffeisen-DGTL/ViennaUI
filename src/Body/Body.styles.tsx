import styled, { createGlobalStyle } from 'styled-components';
import { body } from 'vienna.ui-theme';
import { getPresets } from '../Utils/styling';

const presets = getPresets(
    body,
    'body'
)({
    base: null,
});

export const Box = styled.div`
    ${presets.base}
    font-feature-settings: normal;
    -webkit-font-smoothing: antialiased;
    height: 100%;
    position: relative;
    display: flex;
    flex-direction: column;
`;

export const GlobalStyles = createGlobalStyle`
    html {
        box-sizing: border-box;
        line-height: 1.15;
        -webkit-text-size-adjust: 100%;
    }

    body {
        margin: 0;
    }

    a {
        background-color: transparent;
    }

    *,
    *::before,
    *::after {
        box-sizing: inherit;
    }

    template {
        display: none;
    }

    [hidden] {
        display: none;
    }
`;
