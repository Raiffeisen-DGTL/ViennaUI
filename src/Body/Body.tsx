import React, { ComponentProps, forwardRef } from 'react';
import { ExecutionProps } from 'styled-components';
import { Box, GlobalStyles, GlobalCustomFonts } from './Body.styles';

export type BodyProps = Partial<ExecutionProps & ComponentProps<typeof Box>>;

export const Body = forwardRef<HTMLDivElement, BodyProps>(({ ...attrs }, ref) => (
    <>
        <GlobalStyles />
        <GlobalCustomFonts />
        <Box {...(attrs as {})} ref={ref} />
    </>
));

Body.displayName = 'Body';
