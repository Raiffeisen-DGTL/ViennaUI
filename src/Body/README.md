# Body

Используется как основной контейнер в приложении для установки типографики, шрифтов **Navigo** и **ALS Hauss**. Можно загружать локальные шрифты, а не с сайта, если прокинуть в Body проп localFonts. Также компонент содержит нормализацию основных стилей.

⚠️ Доступные начертания CSS свойства `font-weight` для **ALS Hauss** — `normal`, `500` и `bold`, для **Navigo** — `normal`.

## Импорт

```
import { Body } from 'vienna-ui';
```

# Body

Используется как основной контейнер в приложении для установки типографики, шрифтов **Navigo** и **ALS Hauss**. Можно загружать локальные шрифты, а не с сайта, если прокинуть в Body проп localFonts. Также компонент содержит нормализацию основных стилей.

⚠️ Доступные начертания CSS свойства `font-weight` для **ALS Hauss** — `normal`, `500` и `bold`, для **Navigo** — `normal`.

```
    <Body>Content</Body>
```



## Шрифты

Компонент `Body` содержит исходные файлы шрифтов в директории `@fcc/ui/src/Body/resources`. Вы можете подключить шрифты вручную в вашем проекте и не использовать компонент `Body`.

###### Шрифты ALS_Hauss

```
@fcc/ui/src/Body/resources/ALS_Hauss-Bold.eot
@fcc/ui/src/Body/resources/ALS_Hauss-Bold.woff
@fcc/ui/src/Body/resources/ALS_Hauss-Bold.woff2
@fcc/ui/src/Body/resources/ALS_Hauss-Medium.eot
@fcc/ui/src/Body/resources/ALS_Hauss-Medium.woff
@fcc/ui/src/Body/resources/ALS_Hauss-Medium.woff2
@fcc/ui/src/Body/resources/ALS_Hauss-Regular.eot
@fcc/ui/src/Body/resources/ALS_Hauss-Regular.woff
@fcc/ui/src/Body/resources/ALS_Hauss-Regular.woff2
```

###### Шрифты Navigo

```
@fcc/ui/src/Body/resources/Navigo-Bold-Web.eot
@fcc/ui/src/Body/resources/Navigo-Bold-Web.woff
@fcc/ui/src/Body/resources/Navigo-Bold-Web.woff2
```

## Возможность скачать шрифты с cdn

Для удобства добавлена возможность скачать шрифты с cdn по ссылке: https://cdn-rf.raiffeisen.ru/ds/fonts/fonts.zip

## Пример загрузки шрифта через src

```
    {() => {
        const fontsCss = `
            @font-face {
                font-family: 'ALS Hauss';
                font-display: swap;
                src: url('https://cdn-rf.raiffeisen.ru/ds/fonts/ALS_Hauss-Regular.eot');
                src: url('https://cdn-rf.raiffeisen.ru/ds/fonts/ALS_Hauss-Regular.woff') format('woff');
                font-weight: normal;
                font-style: normal;
            }
        `;
        return (
            <Body style={{ fontFamily: 'ALS Hauss' }}>
                <style>{fontsCss}</style>
                <p>Этот текст написан шрифтом ALS Hauss.</p>
            </Body>
        );
    }}
```