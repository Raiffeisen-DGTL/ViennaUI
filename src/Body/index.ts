export { Body, type BodyProps } from './Body';
