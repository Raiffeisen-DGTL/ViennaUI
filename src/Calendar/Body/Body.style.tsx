import styled, { css } from 'styled-components';
import { getPresets } from '../../Utils/styling';

const DAY_SIZE = 32;
const CELL_WIDTH = 74;
const CELL_HEIGHT = 56;

const presets = getPresets('calendar', {
    body: null,
    footer: null,
});

const cell = getPresets('calendar.cell', {
    base: null,
    hover: null,
    active: null,
});

export const Box = styled.div`
    box-sizing: border-box;
    width: ${DAY_SIZE * 7 + 16 * 2}px;
    display: flex;
    flex-wrap: wrap;
    ${presets.body}
`;

export const Footer = styled.div`
    display: flex;
    align-items: center;
    justify-content: center;
    ${presets.footer}
`;

interface PropsCalendarCell {
    $isActive?: boolean;
}
export const CalendarCell = styled.div<PropsCalendarCell>`
    width: ${CELL_WIDTH}px;
    height: ${CELL_HEIGHT}px;
    display: flex;
    align-items: center;
    justify-content: center;
    ${cell.base}

    &:hover {
        ${cell.hover}
    }

    ${({ $isActive }) =>
        $isActive &&
        css`
            ${cell.active}
        `}
`;
