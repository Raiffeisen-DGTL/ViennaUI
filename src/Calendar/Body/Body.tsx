/**
 * imports of components
 */
import React, { useMemo } from 'react';
import { Dates, EventDateFunction, RangeDate, StartingWeekDay, ViewMode } from '../types';
import { MonthRangeDays } from '../MonthRangeDays';
import { MonthSingleDays } from '../MonthSingleDays';
import { Button } from '../../Button';
import { useCalendarLocalization } from '../Context';
import { CalendarLocalization } from '../localization';
/**
 * imports of styles
 */
import { Box, CalendarCell, Footer } from './Body.style';

const MONTHS = [
    'january',
    'february',
    'march',
    'april',
    'may',
    'june',
    'july',
    'august',
    'september',
    'october',
    'november',
    'december',
] as const;

interface Props {
    viewMode: ViewMode;
    displayedDate: Date;
    date?: Date | Date[];
    dateStart?: Date;
    dateEnd?: Date;
    maxDate?: Date;
    minDate?: Date;
    ranged?: boolean;
    todayButton?: boolean;
    disabledDates?: Dates;
    weekendDates?: Dates;
    eventDates?: Date[] | EventDateFunction;
    onGoToMonth: (year: number, month: number) => () => void;
    onChangeMonth: (year: number, month: number) => () => void;
    onChangeDate: (nextDate: Date | RangeDate<Date> | Date[]) => void;
    onGoToToday: () => void;
    hasNavigation: boolean;
    startingWeekDay: StartingWeekDay;
    allowMultiple: boolean;
}

export const Body: React.FC<Props> = (props) => {
    const {
        viewMode,
        displayedDate,
        date,
        dateStart,
        dateEnd,
        maxDate,
        minDate,
        ranged,
        disabledDates,
        weekendDates,
        eventDates,
        todayButton,
        onGoToMonth,
        onChangeDate,
        onGoToToday,
        onChangeMonth,
        hasNavigation,
        startingWeekDay,
        allowMultiple,
    } = props;

    const localize = useCalendarLocalization();

    const content = useMemo(() => {
        const displayedYear = displayedDate?.getFullYear();
        const displayedMonth = displayedDate?.getMonth();

        switch (viewMode) {
            case 'year_list': {
                const startYear = Math.floor(displayedYear / 12) * 12;
                const start = startYear <= 0 ? 0 : startYear;
                const end = startYear + 12 >= 3000 ? 3001 : startYear + 12;
                const yearList: JSX.Element[] = [];
                for (let year = start; year < end; year++) {
                    const isActive = date && !allowMultiple && (date as Date).getFullYear() === year;

                    yearList.push(
                        <CalendarCell key={year} $isActive={isActive} onClick={onGoToMonth(year, displayedMonth)}>
                            {year}
                        </CalendarCell>
                    );
                }

                return (
                    <div>
                        <Box>{yearList}</Box>
                    </div>
                );
            }
            case 'month_list': {
                const months = MONTHS.map((month, index: number) => {
                    const isActive = date && !allowMultiple && index === (date as Date).getMonth();
                    const localizedMonth = localize(`ds.calendar.month.${month}` as keyof CalendarLocalization);
                    return (
                        <CalendarCell
                            key={localizedMonth}
                            $isActive={isActive}
                            onClick={
                                hasNavigation ? onGoToMonth(displayedYear, index) : onChangeMonth(displayedYear, index)
                            }>
                            {localizedMonth}
                        </CalendarCell>
                    );
                });

                return (
                    <div>
                        <Box>{months}</Box>
                    </div>
                );
            }
            case 'month':
            default: {
                const month = ranged ? (
                    <MonthRangeDays
                        date={{ start: dateStart, end: dateEnd }}
                        disabledDates={disabledDates}
                        weekendDates={weekendDates}
                        eventDates={eventDates}
                        displayedDate={displayedDate}
                        maxDate={maxDate}
                        minDate={minDate}
                        startingWeekDay={startingWeekDay}
                        onChangeDate={onChangeDate}
                    />
                ) : (
                    <MonthSingleDays
                        date={date}
                        disabledDates={disabledDates}
                        weekendDates={weekendDates}
                        eventDates={eventDates}
                        displayedDate={displayedDate}
                        maxDate={maxDate}
                        minDate={minDate}
                        startingWeekDay={startingWeekDay}
                        allowMultiple={allowMultiple}
                        onChangeDate={onChangeDate}
                    />
                );

                return (
                    <div>
                        <Box>{month}</Box>
                        {todayButton && (
                            <Footer>
                                <Button type='button' design='ghost' onClick={onGoToToday}>
                                    <b>{localize('ds.calendar.body.today')}</b>
                                </Button>
                            </Footer>
                        )}
                    </div>
                );
            }
        }
    }, [viewMode, displayedDate, date, dateStart, dateEnd, minDate, maxDate, disabledDates, allowMultiple]);

    return <>{content}</>;
};
