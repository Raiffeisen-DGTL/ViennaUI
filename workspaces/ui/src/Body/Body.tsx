import React, { HTMLAttributes } from 'react';
import { Box, GlobalStyles, GlobalCustomFonts } from './Body.styles';

export interface BodyProps extends HTMLAttributes<HTMLDivElement> {
    [key: string]: any;
    children: React.ReactNode;
}

export const Body: React.FC<BodyProps> = (props): JSX.Element => {
    const { children, ...attrs } = props;

    return (
        <>
            <GlobalStyles />
            <GlobalCustomFonts />
            <Box {...attrs}>{children}</Box>
        </>
    );
};

Body.displayName = 'Body';
export default Body;
