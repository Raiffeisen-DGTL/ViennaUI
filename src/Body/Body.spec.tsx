import React from 'react';
import { Body } from './Body';

test('Body', () => {
    const snap = snapshot.render(<Body>Hello body!</Body>);
    expect(snap).toMatchSnapshot();
});
