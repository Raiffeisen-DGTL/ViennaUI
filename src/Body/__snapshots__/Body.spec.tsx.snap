// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Body 1`] = `
.c0 {
  font-size: 14px;
  font-family: 'ALS Hauss',Helvetica,'Helvetica New',Arial,sans-serif;
  font-weight: 400;
  font-stretch: normal;
  font-style: normal;
  line-height: 22px;
  letter-spacing: normal;
  color: #2B2D33;
  font-feature-settings: normal;
  -webkit-font-smoothing: antialiased;
  height: 100%;
  position: relative;
  display: flex;
  flex-direction: column;
}

<div
  class="c0"
>
  Hello body!
</div>
`;
